### ISSUE TITLE
Target Display Mode not starting without keyboard

### ISSUE THREAD

**Issue by architector1** [duanefields/VirtualKVM#9]
This report relates to Issue number 2.

I have two iMacs, only the master iMac has a cabled keyboard. 
VirtualKVM on the slave iMac recognizes when the master is connected via Thunderbolt, but it does't toogle to TDM since no keyboard is connected.
Having two keyboards connected all time isn't a real solution because then I can also just use the shortcut on the slave iMac to start TDM.

**Comment by stupudmunkee** [comment]
Yeah, I don't think Duane's code was intended to handle USB keyboards. 

If Apple TDM requires both iMacs to have a keyboard to operate, you'd probably need to report it as a bug or feature request to them.

**Comment by SgtOddball** [comment]
I've found that so long as you've had a bluetooth keyboard paired up to the target Mac then it works just fine. So long as it's recognised that there's a bluetooth keyboard then the keyboard is no longer required and it all works fine so i've found. 

So your best work around is to borrow a bluetooth keyboard and pair it, and then return said keyboard and everything'll work just fine.

**Comment by WarpRat** [comment]
@SgtOddball 

How long have you had that set up? Do you find that you need to repair the keyboard after a certain amount of time? If so do you have an idea of how long that is?

**Comment by SgtOddball** [comment]
Right recent findings are as follows:
1: Yes, the keyboard does need to re-pair after a while so instead I've gone different route and bought an app called 1Keyboard which spoofs the keyboard on a MBP to be a bluetooth keyboard.
2: On MacOS sierra, VirtualKVM doesn't detect when the cables pulled out cleanly so often gets stuck thinking the cable is still present. Closing down and restarting VirtualKVM does the trick (I use remote desktop to sort this. Saves on the extra keyboards).
3: I also don't let the screen sleep instead I've got a screen saver which stops the connection thinking it's still active when the MBP has gone to sleep.
4: Finally I've got an SSH connection set so I can terminal into the iMac and lock the screen remotely. 

Bit longwinded but It solves most of my issues so I'll live with it for now.
